File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.cpp
class Solution {
public:
    char nextGreatestLetter(vector<char>& letters, char target) {
        vector<int> tmp(letters.begin(), letters.end());
        sort(letters.begin(), letters.end());
        int pos = upper_bound(letters.begin(), letters.end(), target)-letters.begin();
        
        if(pos>=letters.size())return tmp[0];
        return letters[pos];
    }
};

File: 0744-find-smallest-letter-greater-than-target/0744-find-smallest-letter-greater-than-target.java
class Solution {
    public char nextGreatestLetter(char[] a, char target) {
        int l=0;
        int r=a.length-1;
        while(l<=r)
        {
            int mid=(l+r)/2;
            if(target<a[mid]) r=mid-1;
            else l=mid+1;
        }
        return a[l%a.length];
    }
}